scripts/okta/okta-login.ts

```diff
@@ -0,0 +1,33 @@
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference path="./window.d.ts" />
+
+import { getRedirectUrl } from './lib/helper';
+
+// These files is the custom TypeScript for the Okta Login page. This includes this file, the helper file, and the window.d.ts file.
+// We use these files to redirect users from the login page to our custom login/welcome page on Gateway.
+// This is so we can use the Okta SDKs, but still use our own custom login pages.
+
+// We use jsdelivr to load this script, the automatic minification reduces the size.
+// e.g. <script src="https://cdn.jsdelivr.net/gh/guardian/gateway@main/scripts/okta/okta-login.min.js"></script>
+
+// If testing out changes for a particular branch you can change the branch it's looking for in the custom HTML itself.
+// e.g. for a branch called `feature-branch-xyz`
+// <script src="https://cdn.jsdelivr.net/gh/guardian/gateway@feature-branch-xyz/scripts/okta/okta-login.min.js"></script>
+// or commit
+// <script src="https://cdn.jsdelivr.net/gh/guardian/gateway@df4557838d25ab7991130acc4cbe92e6ab063e6d/scripts/okta/okta-login.min.js"></script>
+
+// If you make changes to these files, be sure to run `yarn gen:okta-login` to compile the TypeScript and generate the Javascript file, `okta-login.js`.
```

Comment:
is there a way we could make sure that the script is run during our build step perhaps, in case a future developer misses this somehow? Feel like it might be good to shout somewhere

if there's a diff, maybe we could fail the build or something? 🤔

Reply:
My only worry about if there is a diff is that I'm unsure if the generated chunk is deterministic, i.e does it generate exactly the same on every machine?

Which is why I didn't do this.

Reply:
Webpack at least [seems to use a different salt on different machines](https://stackoverflow.com/questions/42955593/how-to-make-webpack-generate-consistent-hashes-across-multiple-builds) so I don't think idempotency can be guaranteed